src/lib/chain/offer/updateOffers.ts

```diff
@@ -0,0 +1,456 @@
+
+  const txsToSign = [
+    populatedTxs.flatMap(({ txsToRunBeforeOthers }) => {
+      return txsToRunBeforeOthers.map(({ tx }) => {
+        return tx;
+      });
+    }),
+    populatedTxs.flatMap(({ txs }) => {
+      return txs.map(({ tx }) => {
+        return tx;
+      });
+    }),
+  ].flat();
+
+  if (txsToSign.length === 0) {
+    commandObj.logToStderr("No changes found for selected offers");
+    return;
+  }
+
+  printOffersToUpdateInfo(populatedTxs);
+
+  if (
+    !(await confirm({
+      message: "Would you like to continue",
+      default: true,
+    }))
+  ) {
+    commandObj.logToStderr("Offers update canceled");
+    return;
+  }
+
+  await assertProviderIsRegistered();
+  await signBatch(txsToSign);
+}
+
+type OnChainOffer = Awaited<
+  ReturnType<typeof filterOffersFoundOnChain>
+>[number];
+
+function filterOffersFoundOnChain(offers: EnsureOfferConfig[]) {
+  const [offersNotFoundOnChain, offersToUpdateFoundOnChain] =
+    splitErrorsAndResults(
+      offers,
+      ({
+        offerInfo: { offerInfo, offerIndexerInfo } = {
+          offerInfo: undefined,
+          offerIndexerInfo: undefined,
+        },
+        offerId,
+        ...rest
+      }) => {
+        return offerId === undefined ||
+          offerInfo === undefined ||
+          offerIndexerInfo === undefined
+          ? { error: { ...rest, offerId } }
+          : { result: { ...rest, offerInfo, offerIndexerInfo, offerId } };
+      },
+    );
+
+  if (offersNotFoundOnChain.length > 0) {
+    commandObj.warn(
+      `Some of the offers are not found on chain:\n${offersNotFoundOnChain
+        .map(({ offerName, offerId }) => {
+          return `${offerName}${offerId === undefined ? "" : ` (${offerId})`}`;
+        })
+        .join(
+          "\n",
+        )}\n\nPlease make sure the offers exist. If the offers don't exist you can create them using '${CLI_NAME} provider offer-create' command`,
+    );
+  }
+
+  return offersToUpdateFoundOnChain;
+}
+
+type PeersOnChain = {
+  computeUnits: ComputeUnit[];
+  peerIdBase58: string;
+  hexPeerId: string;
+}[];
+
+type Txs = { description?: string; tx: ReturnType<typeof populateTx> }[];
+
+function populateUpdateOffersTxs(offersFoundOnChain: OnChainOffer[]) {
+  return Promise.all(
+    offersFoundOnChain.map(async (offer) => {
+      const { offerName, offerId, offerIndexerInfo } = offer;
+
+      const peersOnChain = (await Promise.all(
+        offerIndexerInfo.peers.map(async ({ id, ...rest }) => {
+          return {
+            peerIdBase58: await peerIdHexStringToBase58String(id),
+            hexPeerId: id,
+            ...rest,
+          };
+        }),
+      )) satisfies PeersOnChain;
+
+      const effectorsOnChain = await Promise.all(
+        offerIndexerInfo.effectors.map(({ cid }) => {
+          return cidHexStringToBase32(cid);
+        }),
+      );
+
+      const txsToRunBeforeOthers = (await populatePeersToRemoveTxs(
```

Comment:
Why are these to run before others? What are "others"?

Reply:
yeah because you have to first remove before adding. I will rename so it's clearer